### ISSUE TITLE
shrine of the mermaid fix

### ISSUE THREAD

**Issue by holtrop** [kevinkey/klank#81]
If the user plays Shrine of the Mermaid, do not allow them to select teleport if they are not in a flooded room.

**Comment by ajschwieterman** [comment]
What if the player knows they will end up in a flooded room and wants to bank the teleport once they enter a flooded room?

**Comment by holtrop** [comment]
Oh is that how it's implemented? The way I read the card description, you have to be in a flooded room at the time you buy the device to be able to teleport.

**Comment by ajschwieterman** [comment]
Yep.  If you choose the teleport option, the game saves the teleport as a `shrine_mermaid_teleport`, and once you enter a flooded room, the game will give you the 'teleport' option in the player menu to teleport.

https://github.com/kevinkey/klank/blob/d75ce170c0b94799dee37a7c1789c729b9825c00/card.rb#L149-L155

https://github.com/kevinkey/klank/blob/d75ce170c0b94799dee37a7c1789c729b9825c00/player.rb#L309-L311
